lambda/tests/test_new_account_iam_role.py

```diff
@@ -0,0 +1,253 @@
+        f'"Principal": {{"AWS": "arn:aws:iam::{ACCOUNT_ID}:root"}}, '
+        f'"Effect": "Allow"}}, {{"Action": "sts:AssumeRole", '
+        f'"Principal": {{"AWS": "arn:aws:iam::222222222222:root"}}, '
+        f'"Effect": "Allow"}}]}}'
+    )
+
+
+@pytest.fixture(scope="function")
+def valid_role(iam_client, valid_trust_policy):
+    """Return a valid role.
+
+    This assumes iam_role_create_trust() works properly, but there are
+    tests for that.
+    """
+    session = boto3.Session(profile_name="testing")
+    iam_resource = session.resource("iam")
+
+    role = lambda_func.iam_role_create_trust(
+        iam_resource, iam_client, "TEST_IAM_ROLE_VALID", valid_trust_policy
+    )
+    return role
+
+
+@pytest.fixture(scope="function")
+def monkeypatch_get_account_id(monkeypatch):
+    """Mock get_account_id() to return a fake account ID."""
+
+    def mock_get_account_id(event):  # pylint: disable=unused-argument
+        return ACCOUNT_ID
+
+    monkeypatch.setattr(lambda_func, "get_account_id", mock_get_account_id)
+
+
+def test_missing_profile_and_no_arn(valid_trust_policy):
+    """Test requirement for an assume_role_arn or AWS profile as input."""
+    with pytest.raises(Exception) as exc:
+        lambda_func.main(
+            aws_profile="",
+            role_name="TEST_IAM_ROLE_MISSING_PROFILE",
+            role_permission_policy="ReadOnlyAccess",
+            trust_policy_json=valid_trust_policy,
+        )
+    assert "One of 'assume-role-arn' or 'aws-profile' is required" in str(exc.value)
+
+
+def test_invalid_trust_policy_json():
+    """Test an invalid JSON string for trust_policy_json argument."""
+    with pytest.raises(Exception) as exc:
+        # JSON string is missing a bracket in the 'Statement' field.
+        lambda_func.main(
+            aws_profile="testing",
+            role_name="TEST_IAM_ROLE_INVALID_JSON",
+            role_permission_policy="ReadOnlyAccess",
+            trust_policy_json=(
+                f'{{"Version": "2012-10-17", "Statement": '
+                f'[{{"Action": "sts:AssumeRole", '
+                f'"Principal": {{"AWS": "arn:aws:iam::{ACCOUNT_ID}:root"}}, '
+                f'"Effect": "Allow"}}'
+            ),
+        )
+    assert "'trust-policy-json' contains badly formed JSON" in str(exc.value)
+
+
+def test_main_func_valid_arguments(
+    iam_client, valid_trust_policy
+):  # pylint: disable=unused-argument
+    """Test use of valid arguments for main()."""
+    return_code = lambda_func.main(
+        aws_profile="testing",
+        role_name="TEST_IAM_ROLE_VALID_ARGS",
+        role_permission_policy="ReadOnlyAccess",
+        trust_policy_json=valid_trust_policy,
+    )
+    assert return_code == 0
+    roles = [role["RoleName"] for role in iam_client.list_roles()["Roles"]]
+    assert "TEST_IAM_ROLE_VALID_ARGS" in roles
+
+
+def test_iam_role_create_trust_func_bad_args(
+    iam_client, valid_trust_policy, caplog
+):  # pylint: disable=unused-argument
```

Comment:
Good catch ... I originally was also passing in "aws_credentials" and later realized that was unnecessary if I was passing in the fixture "iam_client".  This is why it's so bad to disable pylint!

Reply:
I did a little more research and came across a couple options that might help here: 

* The decorator `@pytest.mark.usefixtures()` injects the named fixtures without modifying the function signature
* The plugin [`pytest-pylint`](https://github.com/reverbc/pylint-pytest) has logic to handle these idiosyncrasies of pytest

Reply:
`@pytest.mark.usefixtures()` can't be used for fixtures themselves.  So I couldn't use it for the `iam_client()` fixture, for example.

`pytest-pylint` is yet another tool for dependabot to complain about, but the main problem is it not picking up PYTHONPATH.  I can play with it some more, but I have other issues I want to resolve first.

Reply:
Oops, I reversed the parts in the plugin name. The one I linked to is `pylint-pytest`: https://github.com/reverbc/pylint-pytest

Reply:
That would require a change to tardigrade-ci to modify:

1.  `.pylintrc` to specify a loadable plugin 
2. `Makefile` to install `pylint-pytest` when `pylint` is installed.  

Separate task.

Reply:
Hmm, I don't think so. Not since we restructured tardigrade-ci to use the project root as the docker workdir, and began running the tooling directly in the workdir. `pylint` will use the config file present in the project, not the config file from tardigrade-ci. Simply adding the config here, and adding `pylint-pytest` to `tests/requirements.txt` ought to work.

Reply:
I would prefer not to add customized .pylintrc as that defeats the purpose of having a "general" policy towards pylinting.  If the pylintrc customization was truly unique and not needed for other pytest tests, then a customization could be rationalized.  But it would be rare to have pytests that don't use fixtures.  So there's a generalized need here.

Reply:
Changes are ready for review in the `tardigrade-ci` repo to load the `pylint-pytest` plugin in the `.pylintrc` file.